SpellChecker.java

```diff
@@ -11,25 +11,50 @@ public static void main(String[] args) {
 	}
 
 	public static String tail(String str) {
-		// Your code goes here
+			return str.substring(1);
 	}
 
 	public static int levenshtein(String word1, String word2) {
-		// Your code goes here
+		String w1 = word1.toLowerCase();
+		String w2 = word2.toLowerCase();
+		if(w1.length() == 0){
+			return w2.length();
+		}
+		if(w2.length() == 0){
+			return w1.length();
+		}
+		if(w1.charAt(0) == w2.charAt(0)){
+			return levenshtein(tail(w1),tail(w2));
+		}
+		int min = (int)(Math.min(levenshtein(tail(w1), w2),  levenshtein(w1, tail(w2))));
+		return 1 + (int)(Math.min(min, levenshtein(tail(w1), tail(w2))));
```

Comment:
No need casting here
```suggestion
		int min = Math.min(levenshtein(tail(w1), w2),  levenshtein(w1, tail(w2)));
		return 1 + Math.min(min, levenshtein(tail(w1), tail(w2)));
```